controller/src/main/java/org/jboss/as/controller/transform/description/TransformingDescription.java

```diff
@@ -137,14 +137,15 @@ void invokeNext(OperationTransformer.TransformedOperation transformedOperation)
 
     @Override
     public void transformResource(final ResourceTransformationContext ctx, final PathAddress address, final Resource original) throws OperationFailedException {
-        final ModelNode originalModel = TransformationRule.cloneAndProtect(original.getModel());
         final DiscardPolicy discardPolicy = determineDiscardPolicy(ctx, address);
 
         switch (discardPolicy) {
-            case DISCARD_AND_WARN:
             case REJECT_AND_WARN:
                 ctx.getLogger().logRejectedResourceWarning(address, null);
                 return;
+            case DISCARD_AND_WARN:
+                // don't return yet, just log a warning first and then discard
+                ctx.getLogger().logDiscardedResourceWarning(address);
```

Comment:
I would also add a parameter for the host name:
 ctx.getTarget().getHostName()
Apart from that it looks good